src/api/changePassword.js

```diff
@@ -0,0 +1,48 @@
+const router = require("express").Router();
+const nodemailer = require("nodemailer");
+const cors = require("cors");
+const User = require("../models/User");
+const bodyParser = require("body-parser");
+const jwt = require("jsonwebtoken");
+const exjwt = require("express-jwt");
+const sendResponse = require("../utils/sendResponse");
+const { SECRET_TOKEN } = require("../utils/secret-token");
+
+router.post("/changePassword", async function(req, res) {
+  if (!req.body || !req.body.email || !req.body.token || !req.body.password) {
+    sendResponse(
+      res,
+      400,
+      "Request doesn't contain all of: email, token, or password"
+    );
+    return;
+  }
+
+  const user = await User.findOne({ email: req.body.email }).catch(e =>
+    console.log(e)
+  );
+  if (user) {
+    var new_token = jwt.sign(
+      { email: req.body.email, password: req.body.password },
+      SECRET_TOKEN
+    );
+    if (user.email === req.body.email && req.body.token === user.password) {
+      user.password = new_token;
```

Comment:
The password should be the new password that they set, and not the new token.

Reply:
you can also use user.updateOne({_id: user._id},{password: req.body.newPassword})

Reply:
By setting the password to the new password in plaintext, this will conflict with @hungryflamingcat's storing of the token in place of the password.

Reply:
Yup, your absolutely right. Nice catch!